File: src/lib/garage/lastRideHero.ts
import type { Ride } from "@/types";
import { lastRideForBike } from "@/lib/maintenance/summary";
import { honestClimbM } from "@/lib/ride/rideTelemetry";

/** Hero-Zeile: nur echte Fahrten, keine erfundenen Kilometer. */
export function lastRideHeroLine(ride: Ride | undefined): string | null {
  if (!ride) return null;
  const km = ride.distanceM / 1000;
  if (Number.isFinite(km) && km >= 0.05) {
    const climb = honestClimbM(ride.track, ride.elevationGainM);
    const hm = climb >= 10 ? ` · ${climb} hm` : "";
    return `Zuletzt ${km.toFixed(1)} km${hm}`;
  }
  return "Zuletzt unterwegs — ohne GPS-Strecke";
}

export function lastRideHeroLineForBike(
  rides: Ride[],
  bikeId: string
): string | null {
  const ended = rides.filter((r) => Boolean(r.endTime));
  return lastRideHeroLine(lastRideForBike(ended, bikeId));
}

File: mobile/lib/domain/garage/last_ride_hero.dart
import '../ride.dart';
import '../ride/ride_telemetry.dart';

/// Hero-Zeile: nur echte Fahrten, keine erfundenen Kilometer.
String? lastRideHeroLine(RideRecord? ride) {
  if (ride == null) return null;
  if (ride.distanceKm >= 0.05) {
    final climb = honestClimbM(ride.track, ride.elevationM);
    final hm = climb >= 10 ? ' · $climb hm' : '';
    return 'Zuletzt ${ride.distanceKm.toStringAsFixed(1)} km$hm';
  }
  return 'Zuletzt unterwegs — ohne GPS-Strecke';
}

RideRecord? lastEndedRideForBike(List<RideRecord> rides, String bikeId) {
  RideRecord? best;
  for (final r in rides) {
    if (r.bikeId != bikeId) continue;
    if (r.endedAt == null) continue;
    if (best == null || r.startedAt.isAfter(best.startedAt)) best = r;
  }
  return best;
}
